src/features/speedup/components/SpeedUpModal.tsx

```diff
@@ -196,9 +195,19 @@ export const SpeedUpModal = ({
           <Button onClick={onCancel}>Cancel</Button>
 
           <Tooltip title="Speed up transaction">
-            <Button color="primary" disabled={isDisabled} onClick={onSubmit} variant="contained" disableElevation>
-              {isDisabled ? 'Waiting on confirmation in wallet...' : 'Confirm'}
-            </Button>
+            <CheckWallet checkNetwork={!isDisabled}>
+              {(isOk) => (
+                <Button
+                  color="primary"
+                  disabled={!isOk || isDisabled}
+                  onClick={onSubmit}
+                  variant="contained"
+                  disableElevation
+                >
+                  {isDisabled ? 'Waiting on confirmation in wallet...' : 'Confirm'}
```

Comment:
`Waiting on confirmation in wallet...` – that's a long piece of text for a button, how does it look in this state?

Reply:
Changed to a loading spinner like we have it on the execute/sign buttons